### PR TITLE
Ignore warnings reading merged remote branches

### LINKED ISSUE #9462: Deleting remote branch is broken
## Current behaviour

image

## Steps to reproduce

1. Latest master (adbb410d5b65b8d23d8bb71541179cbe573478e8)
2. Right click > delete remote branch or <kbd>DEL</kbd> on a remote branch

## Did this work in previous version of GitExtensions

Yes, not sure where we regressed.

### PR DESCRIPTION
**Pull request by RussKie**
Fixes #9462

## Test methodology 

- manual
- unit tests

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: UnitTests/CommonTestUtils/MockExecutable.cs [hunk 1]
```diff
@@ -133,6 +134,18 @@ public Task<int> WaitForExitAsync()
...
+#pragma warning disable VSTHRD103 // Call async methods when in an async method
+                        string errorOutput = StandardError.ReadToEnd().Trim();
+#pragma warning restore VSTHRD103 // Call async methods when in an async method
+                        if (_exitCode.Value != 0 || errorOutput.Length > 0)
+                        {
+                            ExternalOperationException ex = new(exitCode: _exitCode, innerException: new Exception(errorOutput));
+                            return Task.FromException<int>(ex);
+                        }
+                    }
+
```
Comment (RussKie) [inline UnitTests/CommonTestUtils/MockExecutable.cs:148]: @mstv this is a "poor man" integration tests for https://github.com/gitextensions/gitextensions/pull/9056. This implementation must match the "prod" functionality.

File: UnitTests/CommonTestUtils/MockExecutable.cs [hunk 2]
```diff
@@ -1,47 +1,45 @@
...
 using GitUIPluginInterfaces;
-using JetBrains.Annotations;
 using NUnit.Framework;
 
 namespace CommonTestUtils
 {
     public sealed class MockExecutable : IExecutable
     {
-        private readonly ConcurrentDictionary<string, ConcurrentStack<(string output, int? exitCode)>> _outputStackByArguments = new();
+        private readonly ConcurrentDictionary<string, ConcurrentStack<(string? output, string? error, int? exitCode)>> _outputStackByArguments = new();
```
Comment (gerhardol) [inline UnitTests/CommonTestUtils/MockExecutable.cs:18]: GetOutputLines() combines StandardOutput and StandardError already, so this adds another level to the tests that do not really exist.
GetOutputLines() should be changed to a standard method instead (possibly a "Raw" variant of Execute) and a helper function for combining the output lines.

**Review by gerhardol — COMMENTED** [review]
-1
Code changes are fine, but I do not want the test changes merged.

**Comment by RussKie** [comment]
The mock must reflect the concrete implementation, otherwise we'd continue
to regress making changes to the Process implementation.

How else would you test the current regression?

**Comment by gerhardol** [comment]
> The mock must reflect the concrete implementation, otherwise we'd continue to regress making changes to the Process implementation. How else would you test the current regression?

Rewrite the IExecutable part and separate the GetOutput method and have a separate test of GetOutput.

File: GitCommands/Git/GitModule.cs [hunk 3]
```diff
@@ -2964,7 +2964,7 @@ public IReadOnlyList<string> GetMergedRemoteBranches()
             var remotes = GetRemoteNames();
 
             return _gitExecutable
-                .GetOutputLines(GitCommandHelpers.MergedBranchesCmd(includeRemote: true))
+                .GetOutputLines(GitCommandHelpers.MergedBranchesCmd(includeRemote: true), throwOnErrorOutput: false)
```
Comment (mstv) [inline GitCommands/Git/GitModule.cs:2967]: Your broken HEAD ref is not a sufficient rationale for suppressing the error handling.

**Review by mstv — COMMENTED** [review]
-2

**Comment by RussKie** [comment]
The broken HEAD is irrelevant in this case. We've broken the UX, which
worked for years.

Please have it fixed.

**Comment by gerhardol** [comment]
Suppressing the error is probably OK in the short term if the parsing still works. That is should be seen as a temporary measure that may hide errors. 

The real fix is to not include stderr in GetOutputLines(). Some rewriting required.
For GetOutput() there is a simple alternative to use Execute(), but really all of these should be rewritten too.

Do not adapt the tests to work with the broken implementation.

The question when the Executable interface is fine is what to consider an error and report to the user:
* Everything (as in master)
* All error exit and all but some filtered outputs
* All error exit and some filtered outputs
* Only error exit
* None (as in this PR and in 3.5)

**Comment by RussKie** [comment]
@msftbot merge in 24 hours

**Comment by gerhardol** [comment]
I believe this issue can be closed after #9562

**Pull request closed without merging**